pkg/v1/cli/pluginmanager/manager.go

```diff
@@ -189,6 +176,33 @@ func AvailablePlugins(serverName string) ([]plugin.Discovered, error) {
 	return availablePlugins, nil
 }
 
+func availablePluginsFromStandaloneAndServerPlugins(discoveredServerPlugins, discoveredStandalonePlugins []plugin.Discovered) []plugin.Discovered {
+	availablePlugins := discoveredServerPlugins
+
+	for i := range discoveredStandalonePlugins {
+		matchIndex := -1
+		for j := range availablePlugins {
+			if discoveredStandalonePlugins[i].Name == availablePlugins[j].Name {
+				matchIndex = j
+				break
+			}
+		}
+
+		// Add the standalone plugin to available plugins if it doesn't exist in the serverPlugins list
+		// OR
+		// Current standalone discovery is of type 'local'
+		// We are overriding the discovered plugins that we got from server in case of 'local' discovery type
+		// to allow developers to use the plugins that are built locally and not returned from the server
+		// This local discovery is only used for development purpose and should not be used for production
+		if config.GetDefaultStandaloneDiscoveryType() == common.DiscoveryTypeLocal && matchIndex >= 0 {
+			availablePlugins[matchIndex] = discoveredStandalonePlugins[i]
+		} else if matchIndex < 0 {
+			availablePlugins = append(availablePlugins, discoveredStandalonePlugins[i])
+		}
```

Comment:
Almost every time I see `else` or `else if`, it hints that the code could be simpler. For example, here it could look like: 

```go
if matchIndex < 0 {
	availablePlugins = append(availablePlugins, discoveredStandalonePlugins[i])
	continue
}
if isLocal { // matchIndex >= 0 is guaranteed here
	availablePlugins[matchIndex] = discoveredStandalonePlugins[i]
}
```